### PR TITLE
Add ReadableStream to Jest integration.

### PR DESCRIPTION
**Pull request by kenjim83**
From https://github.com/mswjs/msw/discussions/1934

### CODE REVIEW

File: src/content/docs/shared/jest-missing-globals.mdx [hunk 1]
```diff
@@ -24,11 +24,12 @@ Create a `jest.polyfills.js` file next to your `jest.config.js` with the followi
  * you don't want to deal with this.
  */
 
-const { TextDecoder, TextEncoder } = require('node:util')
+const { TextDecoder, TextEncoder, ReadableStream } = require('node:util')
```
Comment (kettanaito) [inline src/content/docs/shared/jest-missing-globals.mdx:27]: `node:util` doesn't export `ReadableStream`, `node:streams/web` does.

**Comment by kettanaito** [comment]
Hi, @kenjim83. Reaching back to you to provide additional context on `ReadableStream` in Jest. 

As I have [since learned](https://github.com/mswjs/msw/issues/1931#issuecomment-1890461324), the JSDOM environment in Jest doesn't rely on the `jest-environment-node` (which is a rather big oversight), and instead uses `core-js` to polyfill already existing Node.js globals, like `ReadableStream`. The `core-js` module just throws whenever it tries to call `structuredClone()` on a `ReadableStream` instance because it's not implemented there. 

TL;DR There's no way to get `ReadableStream` to work in Jest unless Jest (or `core-js`) itself addresses the issue. No polyfills on your end will be respected by Jest (one has to polyfill `structuredClone()` and Node.js doesn't expose its implementation, it's an internal C++ binding).
